assembly/src/tests.rs

```diff
@@ -194,6 +194,76 @@ fn multiple_constants_push() {
     assert_eq!(expected, format!("{program}"));
 }
 
+#[test]
+fn constant_numeric_expression() {
+    let assembler = super::Assembler::default();
+    let source = "const.TEST_CONSTANT=11-2+4*(12-(10+1))+9+8/4*2 \
+    begin \
+    push.TEST_CONSTANT \
+    end \
+    ";
+    let expected = "\
+    begin \
+        span \
+            push(26) \
+        end \
+    end";
+    let program = assembler.compile(source).unwrap();
+    assert_eq!(expected, format!("{program}"));
+}
+
+#[test]
+fn constant_alphanumeric_expression() {
+    let assembler = super::Assembler::default();
+    let source = "const.TEST_CONSTANT_1=(18-1+10)*6-((13+7)*2) \
+    const.TEST_CONSTANT_2=11-2+4*(12-(10+1))+9
+    const.TEST_CONSTANT_3=(TEST_CONSTANT_1-(TEST_CONSTANT_2+10))/5+3
+    begin \
+    push.TEST_CONSTANT_3 \
+    end \
+    ";
+    let expected = "\
+    begin \
+        span \
+            push(21) \
+        end \
+    end";
+    let program = assembler.compile(source).unwrap();
+    assert_eq!(expected, format!("{program}"));
+}
+
+#[test]
+fn constant_field_division() {
+    let assembler = super::Assembler::default();
+    let source = "const.TEST_CONSTANT=(16//4)/(2//2)*4 \
```

Comment:
nit: could we try integer division on values that don't divide evenly? e.g., `17/4` should still give `4`.